### ISSUE TITLE
Generator/Deterministic generation not working for Karlo/unCLIP 

### ISSUE THREAD

**Issue by apolinario** [huggingface/diffusers#4591]
### Describe the bug

Regardless of sending a generator or trying to set a torch seed in the environment, each unCLIP generation makes a different image.

### Reproduction

Setup
```py
import torch
from diffusers import UnCLIPPipeline

pipe = UnCLIPPipeline.from_pretrained("kakaobrain/karlo-v1-alpha", torch_dtype=torch.float16)
pipe = pipe.to("cuda")
generator = torch.Generator(device="cuda").manual_seed(42)
# I also tried generator = [torch.Generator(device="cuda").manual_seed(42)] with same outcome
```
Pipeline:
```py
images = pipe("a mecha robot in a favela", generator=generator).images
images[0]
```
| ![image1](https://github.com/huggingface/diffusers/assets/788417/734cd89f-a544-4b0c-803c-3c4fc52bb4ec) | ![image2](https://github.com/huggingface/diffusers/assets/788417/7d686d17-8821-4234-9cb1-451170641b35) |
|:---:|:---:|
| Generated first | Generated again |

Running this will get me a different image all the time on the same hardware. And even trying all environment seeding techniques, I couldn't get this pipeline to behave deterministically
```py
  seed = 42
  torch.manual_seed(seed)
  torch.cuda.manual_seed(seed)
  torch.backends.cudnn.deterministic = True
  torch.backends.cudnn.benchmark = False
  np.random.seed(seed)
  random.seed(seed)
```

### System Info

`diffusers` == 0.19.3

### Who can help?

cc @williamberman

**Comment by apolinario** [comment]
By doing some poking around the pipeline and trying to debug things, I think the issue is that the `prior` is behaving non-deterministically

For the same prompt with the same generator, the `prior` seems to be generating a different image embedding for the same prompt every time.

**Comment by sayakpaul** [comment]
The generator needs to be reinitialized as it's an advanced one similar to: https://github.com/huggingface/diffusers/issues/4055. 

I was able to perform deterministic generation: https://colab.research.google.com/gist/sayakpaul/5e501007917258f9bc9ace639a000e45/scratchpad.ipynb.
